### PR TITLE
EDGECLOUD-4807: Update sample.

### PR DESCRIPTION
**Pull request by lgarner**
Update sample for R3, EdgeEvents, minimally edited from SDK.

### CODE REVIEW

File: android/EmptyMatchEngineMavenApp/app/src/androidTest/java/com/mobiledgex/emptymatchengineapp/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019 MobiledgeX, Inc. All rights and licenses reserved.
+ * Copyright 2018-2020 MobiledgeX, Inc. All rights and licenses reserved.
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/androidTest/java/com/mobiledgex/emptymatchengineapp/ExampleInstrumentedTest.java:2]: 2021

**Review by franklin-huang-mobiledgex — APPROVED** [review]

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/SettingsActivity.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019 MobiledgeX, Inc. All rights and licenses reserved.
+ * Copyright 2018-2020 MobiledgeX, Inc. All rights and licenses reserved.
```
Comment (brucearmstrong) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/SettingsActivity.java:2]: 2021

**Review by brucearmstrong — APPROVED** [review]
Androidx: Welcome to the future.
This is a good set of demo code.

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/SettingsActivity.java (hunk 2 above)
Comment (lgarner) [reply android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/SettingsActivity.java]: Done.

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java [hunk 3]
```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
...
+
+                    // Our example server might not like random connections to non-existing /test.
+                    String api = serverPort == knownPort ? "/test" : "";
+                    Response response = null;
+                    try {
+                        Request request = new Request.Builder()
+                                .url("http://" + host + ":" + serverPort + api)
+                                .build();
+                        response = client.newCall(request).execute();
+                        someText += "[Test Server response: " + response.toString() + "]";
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java:565]: missing \n

sdk test (originally as automation-sdk-porttest)   might be useful here.
Screen Shot 2021-07-01 at 6 56 18 PM
https://mobiledgex.atlassian.net/wiki/spaces/SWDEV/pages/1615101953/SDK+Test+App

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java [hunk 4]
```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
...
+                    } catch (IOException | IllegalStateException e) {
+                        someText += "[Error connecting to host: " + host + ", port: " + serverPort + ", api: " + api + ", Reason: " + e.getMessage() + "]";
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java:567]: missing \n

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java [hunk 5]
```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
...
+                    portListStr += String.format(Locale.getDefault(), appPortFormat,
+                            aPort.getProto().getNumber(),
+                            aPort.getInternalPort(),
+                            aPort.getPublicPort(),
+                            aPort.getEndPort());
+
+                    //String l7Url = mMatchingEngine.getAppConnectionManager().createUrl(closestCloudlet, aPort, aPort.getPublicPort(), "http", null);
+
+                    String host = aPort.getFqdnPrefix() + closestCloudlet.getFqdn();
+                    int knownPort = 8008;
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java:552]: I think 8008 is CV, You can use any of the ports here https://mobiledgex.atlassian.net/wiki/spaces/SWDEV/pages/1615101953/SDK+Test+App

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java [hunk 6]
```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
...
-                        if (!locationVerificationAllowed) {
-                            someText += " Reason: Enhanced location is disabled.\n";
+                    }
+                    portListStr += String.format(Locale.getDefault(), appPortFormat,
+                            aPort.getProto().getNumber(),
+                            aPort.getInternalPort(),
+                            aPort.getPublicPort(),
+                            aPort.getEndPort());
+
+                    //String l7Url = mMatchingEngine.getAppConnectionManager().createUrl(closestCloudlet, aPort, aPort.getPublicPort(), "http", null);
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java:549]: This is still here as an example?

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java [hunk 7]
```diff
@@ -273,176 +405,268 @@ public void onComplete(Task<Location> task) {
...
-                    } else {
-                        someText = "Cannot create request object.\n";
-                        if (!locationVerificationAllowed) {
-                            someText += " Reason: Enhanced location is disabled.\n";
+                    }
+                    portListStr += String.format(Locale.getDefault(), appPortFormat,
+                            aPort.getProto().getNumber(),
+                            aPort.getInternalPort(),
+                            aPort.getPublicPort(),
+                            aPort.getEndPort());
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java:547]: Is it end port or  Path Prefix?

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/FirstTimeUseActivity.java [hunk 8]
```diff
@@ -152,3 +161,4 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
         mRpUtil.onRequestPermissionsResult(this, requestCode, permissions, grantResults);
     }
 }
+//! [first_time_use_example]
```
Comment (ah1053) [inline android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/FirstTimeUseActivity.java:164]: Extra line at EOF

File: android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java (hunk 5 above)
Comment (ah1053) [reply android/EmptyMatchEngineMavenApp/app/src/main/java/com/mobiledgex/emptymatchengineapp/MainActivity.java]: Garner is the port here is correct?

**Pull request merged**